From e62d76f034c4e9a3d1c3aeb0e9687605d198e351 Mon Sep 17 00:00:00 2001
From: Dan Liang <dan.liang@intel.com>
Date: Tue, 30 Apr 2013 14:11:02 +0800
Subject: [PORT FROM MAIN] MPEG4Extractor: Avoid some unexpected rounded situation when calculating sample time

BZ: 48862

Original patches:
R3: 30130 (BZ: 18544)
R1: 7415 (BZ: 739)

Root cause: Normally the sampletime can be divided without remainder,
a correct frame can be select as the thumbnail. But if remainder exists
in unexpected situation and the sample time will be rounded, the calculated
thumbnail sample time will smaller than the orignail and the framework will
select the previous key frame other then the correct one.

Change-Id: Icfd9400f64bf584d8cf8d33fed61431402d2ee28
Orig-Change-Id: I0d1be4cb8a8f8a1c20bf69dd6faf9af7c8d1eb33
Signed-off-by: Weian Chen <weian.chen@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
---
 media/libstagefright/MPEG4Extractor.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/media/libstagefright/MPEG4Extractor.cpp b/media/libstagefright/MPEG4Extractor.cpp
index 7d4bb23..796323d 100644
--- a/media/libstagefright/MPEG4Extractor.cpp
+++ b/media/libstagefright/MPEG4Extractor.cpp
@@ -438,7 +438,7 @@ sp<MetaData> MPEG4Extractor::getTrackMetaData(
                             &sampleTime) == OK) {
                     track->meta->setInt64(
                             kKeyThumbnailTime,
-                            ((int64_t)sampleTime * 1000000) / track->timescale);
+                            ((((int64_t)sampleTime * 10000000) / track->timescale) + 9) / 10);
                 }
             }
         }
-- 
1.7.4.1

